api/v1/storage_autoscaler.go

```diff
@@ -0,0 +1,105 @@
+	TimeoutSeconds int `json:"timeoutSeconds,omitempty"`
+}
+
+type StorageAutoScalerPhase string
+
+const (
+	StorageAutoScalerPhaseNotStarted StorageAutoScalerPhase = "NotStarted"
+	StorageAutoScalerPhaseInProgress StorageAutoScalerPhase = "InProgress"
+	StorageAutoScalerPhaseSucceeded  StorageAutoScalerPhase = "Succeeded"
+	StorageAutoScalerPhaseFailed     StorageAutoScalerPhase = "Failed"
+)
+
+// StorageAutoScalerStatus defines the observed state of StorageAutoScaler
+type StorageAutoScalerStatus struct {
+	// Phase describes the Phase of StorageAutoScaler
+	// +kubebuilder:validation:Enum=NotStarted;InProgress;Failed;Succeeded
+	Phase StorageAutoScalerPhase `json:"phase,omitempty"`
+	// Conditions represents the latest available observations of an object's state
+	// +kubebuilder:validation:Optional
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
+	// LastExpansionStartTime is the time stamp of the last run start of the storage scaling
+	LastExpansionStartTime metav1.Time `json:"lastExpansionStartTime,omitempty"`
+	// LastExpansionCompletionTime is the time stamp of the last run completion of the storage scaling
+	LastExpansionCompletionTime metav1.Time `json:"lastExpansionCompletionTime,omitempty"`
+	// The start OSD size is the original size of the OSDs before the expansion in progress is completed.
+	// After the expansion is completed, this would be updated to the expected OSD size.
+	// Used for vertical scaling of OSDs.
+	StartOsdSize resource.Quantity `json:"startOsdSize,omitempty"`
+	// The ExpectedOsdSize is the size that the auto-expansion has decided to set.
+	// This will be set on the storageCLuster CR as the desired size of the OSDs.
+	// Used for vertical scaling of OSDs.
+	ExpectedOsdSize resource.Quantity `json:"expectedOsdSize,omitempty"`
+	// The start OSD count is the original count of the OSDs before the expansion in progress is completed.
+	// After the expansion is completed, this would be updated to the expected OSD count.
+	// Used for horizontal scaling of OSDs.
+	StartOsdCount uint16 `json:"startOsdCount,omitempty"`
+	// The Expected OSD count is the count that the auto-expansion has decided to set.
+	// This will be set on the storageCluster CR as the desired count of the OSDs.
+	// Used for horizontal scaling of OSDs.
+	ExpectedOsdCount uint16 `json:"expectedOsdCount,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+// +kubebuilder:printcolumn:name="Age",type=date,JSONPath=.metadata.creationTimestamp, description="Age"
+// +kubebuilder:printcolumn:name="Phase",type=string,JSONPath=.status.phase,description="Current Phase"
+// +kubebuilder:printcolumn:name="LastRunTimeStamp",type=date,JSONPath=.status.lastRunTimeStamp,description="Last Run Time Stamp"
+// +operator-sdk:csv:customresourcedefinitions:displayName="Storage Auto Scaling"
+
+// StorageAutoScaler represents the automatic storage scaling for storage cluster.
+type StorageAutoScaler struct {
```

Comment:
IMO the type name would be better without the "r" suffix, in other words search and replace `StorageAutoScaler` --> `StorageAutoScale`. But it's not important to debate this detail, if you prefer the "r".
```suggestion
type StorageAutoScale struct {
```

Reply:
I think StorageAutoscaler is more generic in terms of  referring a tool,

And its not a hard change anyway to revert. but most of code its already been used like in the UI and alert team codebase, so better to keep this